app/components/Wallet/useWallet.tsx

```diff
@@ -93,5 +119,96 @@ export const useWallet = () => {
 		if (response.status === "success") setNetwork(newNetwork);
 	}, []);
 
-	return { isConnected, balance, network, connectWallet, disconnectWallet, switchNetwork };
+	const fetchUTXOs = useCallback(async (address: string): Promise<UTXO[]> => {
+		try {
+			const response = await axios.get(`${MEMPOOL_API}/address/${address}/utxo`);
+			return response.data.map((utxo: UTXO) => ({
+				txid: utxo.txid,
+				vout: utxo.vout,
+				value: utxo.value,
+				scriptPubKey: utxo.scriptpubkey,
+			}));
+		} catch (error) {
+			throw new Error(`Failed to fetch UTXOs: ${error}`);
+		}
+	}, []);
+
+	const fetchValidateAddress = useCallback(async (address: string): Promise<ValidateAddressI> => {
+		try {
+			const response = await axios.get(`${MEMPOOL_API}/v1/validate-address/${address}`);
+			return response.data;
+		} catch (error) {
+			throw new Error(`Failed to fetch UTXOs: ${error}`);
+		}
+	}, []);
+
+	const sendTxn = useCallback(
+		async (sendAmount: number, fee: number, opReturnInput: string) => {
+			// bitcoin wallet address
+			const bitcoinAddress = addressInfo.find((adr) => adr.purpose === AddressPurpose.Payment);
+			if (!bitcoinAddress) return;
+			// fetch utxos
+			const utxos: UTXO[] = await fetchUTXOs(bitcoinAddress.address);
+			if (!utxos?.length) return;
```

Comment:
we should add a log, and notify user that the inputs are wrong

Reply:
Done. We need better notification handling in our app. We don't have at the moment. I will create a task for this. For now, I have just thrown the error.

Reply:
Please add TODO in the code and link the task.

Reply:
Done